Core/MMKV_IO.cpp

```diff
@@ -603,6 +619,11 @@ bool MMKV::setDataForKey(MMBuffer &&data, MMKVKey_t key, bool isDataHolder) {
                     retain_key(key);
                 }
             }
+            if (onlyOneKey) {
+                // recalulate CRC indirectly, or call recaculateCRCDigestWithIV() directly
+                auto preparedData = m_crypter ? prepareEncode(*m_dicCrypt) : prepareEncode(*m_dic);
+                doFullWriteBack(std::move(preparedData), nullptr, false);
```

Comment:
This hack is confusing & misleading.

Reply:
At first, I thought of adding function just like `overrideDataWithKey`,  not only for this case, but also for every key which new data(value) is no longer than previous data(value): it is absolutely OK to avoid appending of file, just override, so the file expands slower.

But I am worried about calculating CRC files every time which may be time consuming.

I will dig into this as you mentioned. good ideas!

Reply:
It's bad idea to make it a general method for other keys that size doesn't grow. It would create holes in the file.